src/mul.cpp

```diff
@@ -312,18 +312,20 @@ void Mul::dict_add_term_new(const Ptr<RCP<const Number>> &coef, map_basic_basic
                 }
             }
         } else if (is_a<Rational>(*it->second)) {
-            mpz_class q, r, num, den;
-            num = rcp_static_cast<const Rational>(it->second)->i.get_num();
-            den = rcp_static_cast<const Rational>(it->second)->i.get_den();
-            // Here we make the exponent postive and a fraction between
-            // 0 and 1.
-            if (num > den || num < 0) {
-                mpz_fdiv_qr(q.get_mpz_t(), r.get_mpz_t(), num.get_mpz_t(),
-                    den.get_mpz_t());
+            if (is_a<Integer>(*t) || is_a<Rational>(*t)) {
```

Comment:
We automatically expand when the exponent is integer. 
When I change it, I hit this not implemented error,
https://github.com/sympy/csympy/blob/master/src/complex.h#L271

I can move the logic of `https://github.com/sympy/csympy/blob/master/src/pow.cpp#L142` to `Complex::pow` and send a commit. Should I do it in this PR or send a new one?